docs/plutus-ir/plutus-ir.md

```diff
@@ -0,0 +1,193 @@
+# Plutus IR design proposal
+
+## Syntax
+
+The syntax is mostly as in Plutus Core. Where we modify the syntax of Plutus Core we will
+list only the additions and removals, marking them explicitly with `+` or `-`.
+We also require syntax with lists of subparts. We denote a list of `t`-productions with `t..`.
+
+```
+VarDecl VD      := (vardecl x T)
+TyVarDecl TVD   := (tyvardecl alpha K)
+
+Type R, S, T    := - (fix alpha T)
+Term L, M, N    := + (let BK B.. M)
+
+BiningKind BK   := (nonrec)
+                   (rec)
+Binding B       := (termbind VD L)
+                   (typebind TVD T)
+                   (datatypebind D)
+
+Datatype D      := (datatype
+                      TVD   # the type variable for the datatype
```

Comment:
I'd prefer to use `--` for comments (will the language have comments at all?)

Reply:
No, this is just for the design doc.